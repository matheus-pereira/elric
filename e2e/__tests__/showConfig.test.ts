/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {tmpdir} from 'os';
import * as path from 'path';
import {wrap} from 'elric-snapshot-serializer-raw';
import {cleanup, writeFiles} from '../Utils';
import runelric from '../runelric';

const DIR = path.resolve(tmpdir(), 'show-config-test');

beforeEach(() => cleanup(DIR));
afterEach(() => cleanup(DIR));

test('--showConfig outputs config info and exits', () => {
  writeFiles(DIR, {
    '__tests__/test.test.js': `test('test', () => {});`,
    'package.json': JSON.stringify({elric: {environment: 'node'}}),
  });

  let {stdout} = runelric(DIR, [
    '--showConfig',
    '--no-cache',
    // Make the snapshot flag stable on CI.
    '--updateSnapshot',
  ]);

  stdout = stdout
    .replace(/\\\\node_modules\\\\/g, 'node_modules')
    .replace(/\\\\\.pnp\\\\\.\[\^[/\\]+\]\+\$/g, '<<REPLACED_PNP_PATH>>')
    .replace(/\\\\(?:([^.]+?)|$)/g, '/$1')
    .replace(/"cacheDirectory": "(.+)"/g, '"cacheDirectory": "/tmp/elric"')
    .replace(/"name": "(.+)"/g, '"name": "[md5 hash]"')
    .replace(/"version": "(.+)"/g, '"version": "[version]"')
    .replace(/"maxWorkers": (\d+)/g, '"maxWorkers": "[maxWorkers]"')
    .replace(/"\S*show-config-test/gm, '"<<REPLACED_ROOT_DIR>>')
    .replace(/"\S*\/elric\/packages/gm, '"<<REPLACED_elric_PACKAGES_DIR>>');

  expect(wrap(stdout)).toMatchSnapshot();
});
