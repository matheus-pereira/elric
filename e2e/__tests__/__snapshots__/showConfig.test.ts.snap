// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`--showConfig outputs config info and exits 1`] = `
{
  "configs": [
    {
      "automock": false,
      "cache": false,
      "cacheDirectory": "/tmp/elric",
      "clearMocks": false,
      "coveragePathIgnorePatterns": [
        "/node_modules/"
      ],
      "cwd": "<<REPLACED_ROOT_DIR>>",
      "detectLeaks": false,
      "detectOpenHandles": false,
      "errorOnDeprecated": false,
      "extensionsToTreatAsEsm": [],
      "extraGlobals": [],
      "forceCoverageMatch": [],
      "globals": {},
      "haste": {
        "computeSha1": false,
        "enableSymlinks": false,
        "forceNodeFilesystemAPI": false,
        "throwOnModuleCollision": false
      },
      "injectGlobals": true,
      "moduleDirectories": [
        "node_modules"
      ],
      "moduleFileExtensions": [
        "js",
        "jsx",
        "ts",
        "tsx",
        "json",
        "node"
      ],
      "moduleNameMapper": [],
      "modulePathIgnorePatterns": [],
      "name": "[md5 hash]",
      "prettierPath": "prettier",
      "resetMocks": false,
      "resetModules": false,
      "restoreMocks": false,
      "rootDir": "<<REPLACED_ROOT_DIR>>",
      "roots": [
        "<<REPLACED_ROOT_DIR>>"
      ],
      "runner": "<<REPLACED_elric_PACKAGES_DIR>>/elric-runner/build/index.js",
      "setupFiles": [],
      "setupFilesAfterEnv": [],
      "skipFilter": false,
      "slowTestThreshold": 5,
      "snapshotSerializers": [],
      "testEnvironment": "<<REPLACED_elric_PACKAGES_DIR>>/elric-environment-node/build/index.js",
      "testEnvironmentOptions": {},
      "testLocationInResults": false,
      "testMatch": [
        "**/__tests__/**/*.[jt]s?(x)",
        "**/?(*.)+(spec|test).[tj]s?(x)"
      ],
      "testPathIgnorePatterns": [
        "/node_modules/"
      ],
      "testRegex": [],
      "testRunner": "<<REPLACED_elric_PACKAGES_DIR>>/elric-circus/runner.js",
      "testURL": "http://localhost",
      "timers": "real",
      "transform": [
        [
          "\\\\.[jt]sx?$",
          "<<REPLACED_elric_PACKAGES_DIR>>/babel-elric/build/index.js",
          {}
        ]
      ],
      "transformIgnorePatterns": [
        "/node_modules/",
        "<<REPLACED_PNP_PATH>>"
      ],
      "watchPathIgnorePatterns": []
    }
  ],
  "globalConfig": {
    "bail": 0,
    "changedFilesWithAncestor": false,
    "collectCoverage": false,
    "collectCoverageFrom": [],
    "coverageDirectory": "<<REPLACED_ROOT_DIR>>/coverage",
    "coverageProvider": "babel",
    "coverageReporters": [
      "json",
      "text",
      "lcov",
      "clover"
    ],
    "detectLeaks": false,
    "detectOpenHandles": false,
    "errorOnDeprecated": false,
    "expand": false,
    "findRelatedTests": false,
    "forceExit": false,
    "json": false,
    "lastCommit": false,
    "listTests": false,
    "logHeapUsage": false,
    "maxConcurrency": 5,
    "maxWorkers": "[maxWorkers]",
    "noStackTrace": false,
    "nonFlagArgs": [],
    "notify": false,
    "notifyMode": "failure-change",
    "onlyChanged": false,
    "onlyFailures": false,
    "passWithNoTests": false,
    "projects": [],
    "rootDir": "<<REPLACED_ROOT_DIR>>",
    "runTestsByPath": false,
    "skipFilter": false,
    "testFailureExitCode": 1,
    "testPathPattern": "",
    "testSequencer": "<<REPLACED_elric_PACKAGES_DIR>>/elric-test-sequencer/build/index.js",
    "updateSnapshot": "all",
    "useStderr": false,
    "watch": false,
    "watchAll": false,
    "watchman": true
  },
  "version": "[version]"
}
`;
